angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -3,397 +3,80 @@ import {
 
 @Component({
   selector: 'sprk-masthead',
   template: `
     <header [ngClass]="getClasses()" role="banner" [attr.data-id]="idString">
-      <div
-        class="sprk-c-Masthead__content sprk-o-Stack__item sprk-o-Stack sprk-o-Stack--split@xxs"
+      <sprk-stack
+        splitAt="extraTiny"
+        isStackItem="true"
+        additionalClasses="sprk-c-Masthead__content"
+        *ngIf="branding"
       >
         <div
-          class="sprk-c-Masthead__menu sprk-o-Stack__item sprk-o-Stack__item--center-column@xxs"
-        >
-          <button
-            class="sprk-c-Menu"
-            type="button"
-            [attr.aria-expanded]="isNarrowNavOpen ? true : false"
-            (click)="toggleNarrowNav($event)"
-            [attr.aria-controls]="narrowNavId"
-          >
-            <span class="sprk-u-ScreenReaderText">Toggle Navigation</span>
-            <svg
-              [ngClass]="{
-                'sprk-c-Icon': true,
-                'sprk-c-Icon--xl': true,
-                'sprk-c-Menu__icon': true,
-                'sprk-c-Menu__icon--open': isNarrowNavOpen
-              }"
-              aria-hidden="true"
-              viewBox="0 0 64 64"
-              xmlns="http://www.w3.org/2000/svg"
-            >
-              <g>
-                <path
-                  class="sprk-c-Menu__line sprk-c-Menu__line--two"
-                  d="m8 32h48"
-                />
-                <path
-                  class="sprk-c-Menu__line sprk-c-Menu__line--one"
-                  d="m8 18.68h48"
-                />
-                <path
-                  class="sprk-c-Menu__line sprk-c-Menu__line--three"
-                  d="m8 45.32h48"
-                />
-              </g>
-            </svg>
-          </button>
-        </div>
-
-        <div
-          class="sprk-c-Masthead__branding sprk-o-Stack__item sprk-o-Stack__item--center-column@xxs"
+          sprkStackItem
+          *ngIf="collapsibleNavDirective"
+          #mastheadMenuContainer
+          class="sprk-c-Masthead__menu sprk-o-Stack__item--center-column@xxs"
         >
-          <a
-            sprkLink
-            *ngIf="!logoRouterLink"
-            [attr.href]="logoHref"
-            variant="unstyled"
-          >
-            <ng-content select="[logo-slot]"></ng-content>
-            <span class="sprk-u-ScreenReaderText">{{
-              logoLinkScreenReaderText
-            }}</span>
-          </a>
-          <a
-            sprkLink
-            *ngIf="logoRouterLink"
-            [routerLink]="logoRouterLink"
-            variant="unstyled"
-          >
-            <ng-content select="[logo-slot-router]"></ng-content>
-            <span class="sprk-u-ScreenReaderText">{{
-              logoLinkScreenReaderText
-            }}</span>
-          </a>
+          <sprk-masthead-nav-collapsible-button
+            [collapsibleNavId]="collapsibleNavDirective.id"
+            [isOpen]="isCollapsibleNavOpen"
```

Comment:
The collapsible button component emits when it is clicked and listen for those clicks on the `collapsibleNavButtonEvent` and run our code to hide or show the sibling collapsible nav directive if its present. This component also has an `isOpen` input that we bind to so that we can control if the icon is animated open or closed.